src/r0/directory/get_public_rooms_filtered.rs

```diff
@@ -55,3 +65,174 @@ pub struct Filter {
     #[serde(skip_serializing_if = "Option::is_none")]
     pub generic_search_term: Option<String>,
 }
+
+/// Information about which networks/protocols from application services on the
+/// homeserver from which to request rooms.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum RoomNetwork {
+    /// Return rooms from the Matrix network.
+    Matrix,
+    /// Return rooms from all the networks/protocols the homeserver knows about.
+    All,
+    /// Return rooms from a specific third party network/protocol.
+    ThirdParty(String),
+}
+
+impl Default for RoomNetwork {
+    fn default() -> Self {
+        RoomNetwork::Matrix
+    }
+}
+
+impl Serialize for RoomNetwork {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let mut state;
+        match *self {
```

Comment:
If you remove the dereference here...

```suggestion
        match self {
```